ProcessMaker/processmaker
# FOUR-5882: Terminate event not closing all open tasks

## Issue & Reproduction Steps

- Import the process 
[5882 4.2.30-RC2.zip](https://github.com/ProcessMaker/processmaker/files/8566404/5882.4.2.30-RC2.zip)

-  Run a new request, you should see "Form Task", submit it
-  Now  3 tasks should be available: Task1, Task2 and Terminate End Event
-  Open "Task1" and submit it
-  Open "Terminate End Event" and submit it

Even when the Terminate End Event is triggered,  Task 2 is  still active. A terminate should close all open tokens of a request.

## Solution
- Added listener that persists closed tokens during a terminate end event.

## How to Test
Follow the steps to reproduce and verify that
- Create a request and submit Task1, Terminate End Event.  All tokens and the request should be completed
- Create another request and submit Terminat End Event. All tokens and the request should be completed

## Related Tickets & Packages
- [https://processmaker.atlassian.net/browse/FOUR-5882](https://processmaker.atlassian.net/browse/FOUR-5882)

## Code Review Checklist
